### PR TITLE
[JIT] Add file:line:graph to graph printout

### PR DESCRIPTION
**Pull request by jamesr66a**
Example:

```
import torch

@torch.jit.script
def foo(x):
    y = torch.neg(x)
    return x - y

print(foo.graph.debug_str())
```

```
graph(%x : Tensor):
  %2 : int = prim::Constant[value=1]()
  %y : Tensor = aten::neg(%x) # demo.py:5:9
  %3 : Tensor = aten::sub(%x, %y, %2) # demo.py:6:12
  return (%3)
```

### CODE REVIEW

File: torch/csrc/jit/ir.cpp [hunk 1]
```diff
@@ -218,7 +218,8 @@ static std::ostream& indent(std::ostream& out, size_t level) {
 std::ostream& Node::print(
     std::ostream& out,
     size_t level,
-    std::vector<const Node*>* groups) const {
+    std::vector<const Node*>* groups,
+    bool debug) const {
```
Comment (suo) [inline torch/csrc/jit/ir.cpp]: can we use a more descriptive name than debug? Like, printSourceLocations?

**Review by suo — APPROVED** [review]
lgtm, some naming nits

File: torch/csrc/jit/ir.cpp (hunk 1 above)
Comment (jamesr66a) [reply torch/csrc/jit/ir.cpp]: I wanted to leave this general in case we wanted to add more info in the future but sure

Comment (zdevito) [reply torch/csrc/jit/ir.cpp]: I feel like this version should be the default and we have something else that omits it if we want to parse it.

File: torch/csrc/jit/python_ir.cpp [hunk 2]
```diff
@@ -219,10 +219,10 @@ void initPythonIRBindings(PyObject* module_) {
             return ss.str();
           })
       .def(
-          "debug_str",
-          [](Graph& g) -> std::string {
+          "str",
```
Comment (eellison) [inline torch/csrc/jit/python_ir.cpp]: i think this should have a default value

**Comment by ezyang** [comment]
The CUDA failures on this diff are real:

```
Jun 01 04:53:33 ======================================================================
Jun 01 04:53:33 FAIL: test_kernel_cache_multi_gpu (__main__.TestFuser)
Jun 01 04:53:33 ----------------------------------------------------------------------
Jun 01 04:53:33 Traceback (most recent call last):
Jun 01 04:53:33   File "/var/lib/jenkins/workspace/test/common_utils.py", line 358, in wrapper
Jun 01 04:53:33     method(*args, **kwargs)
Jun 01 04:53:33   File "/var/lib/jenkins/workspace/test/common_utils.py", line 149, in wrapper
Jun 01 04:53:33     fn(*args, **kwargs)
Jun 01 04:53:33   File "/var/lib/jenkins/workspace/test/test_jit.py", line 233, in wrapper
Jun 01 04:53:33     fn(*args, **kwargs)
Jun 01 04:53:33   File "test_jit_fuser.py", line 642, in test_kernel_cache_multi_gpu
Jun 01 04:53:33     self.assertEqual(new_cache_size - prev_cache_size, 1)
Jun 01 04:53:33   File "/var/lib/jenkins/workspace/test/common_utils.py", line 528, in assertEqual
Jun 01 04:53:33     super(TestCase, self).assertLessEqual(abs(x - y), prec, message)
Jun 01 04:53:33 AssertionError: 3 not less than or equal to 1e-05 : 
Jun 01 04:53:33 
Jun 01 04:53:33 ----------------------------------------------------------------------
```

**Comment by jamesr66a** [comment]
@ezyang  i can write a fix real quick

**Pull request closed without merging**